### PR TITLE
Bootstrapped Django 2.0.

### PR DESCRIPTION
**Pull request by timgraham**
- [x] Bump  version
- [x] Add stub release notes
- [x] Remove deprecated features
- [x] Increase PBKDF2 iterations ([discussion](https://groups.google.com/d/topic/django-developers/Qab-hRG-SKs/discussion))

This branch won't do any cleanups for dropping Python 2 compatibility.

(to be merged after the stable/1.11.x branch is created in mid-January following Django 1.11 alpha)

### CODE REVIEW

**Comment by timgraham** [comment]
@knbk, I'm not sure if I'm track with properly fixing the `urlpatterns_reverse` tests. It's not clear to me some tests should be adapted or if they can be removed. If you have some time to work on it or advise, that would be great! Here's where I left off: http://dpaste.com/2CA6A67

**Comment by knbk** [comment]
@timgraham I plan to take a look at this somewhere this or next week.

**Comment by knbk** [comment]
I've created https://github.com/django/django/pull/7829 to remove the app_name argument.

**Comment by timgraham** [comment]
@claudep, I'd appreciate if you have time to review this, especially the GIS stuff. To be merged sometime next week if all goes well.

**Comment by claudep** [comment]
Looks good, after a quick look. Tests seem happy!

File: tests/gis_tests/geoapp/tests.py [hunk 1]
```diff
@@ -157,17 +155,17 @@ def test_geometryfield(self):
...
+        "Database functions on inherited Geometry fields."
         # Creating a Pennsylvanian city.
         PennsylvaniaCity.objects.create(name='Mansfield', county='Tioga', point='POINT(-77.071445 41.823881)')
 
         # All transformation SQL will need to be performed on the
         # _parent_ table.
-        qs = PennsylvaniaCity.objects.transform(32128)
+        qs = PennsylvaniaCity.objects.annotate(new_point=functions.Transform('point', srid=32128))
 
         self.assertEqual(1, qs.count())
```
Comment (timgraham) [inline tests/gis_tests/geoapp/tests.py]: This fails on Oracle with `ORA-22901: cannot compare VARRAY or LOB attributes of an object type`.

**Pull request merged**